luigi/contrib/mrjob.py

```diff
@@ -0,0 +1,288 @@
+    if path.startswith('s3://') or path.startswith('hdfs://'):
+        remote = True
+    return path, remote
+
+
+class MRJobTask(Task):
+    ''' Class that runs a MRJob from a Luigi workflow.
+    '''
+    __metaclass__ = abc.ABCMeta
+
+    @property
+    def job_cls(self):
+        '''
+        Override this to return the right mrjob class (not instance)
+        '''
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def input(self):
+        """ The inputs supported are as follows:
+            - one input
+            - a list of inputs
+            - a dict that contains a list of inputs under the key "input"
+
+            These will be parsed into command line input directives for MRJob.
+            The last one is intended so that it is possible for jobs to depend
+            on files that won't be passed as regular input, e.g. lookup
+            databases, pickled classifier objects or similar that will be
+            distributed to each mapper node separately using MRJob's included
+            batteries.
+
+            Note that most MRJob runners don't really handle moving things to
+            and fro between filesystems. EMR and Hadoop runners can move input
+            data from local file system to S3 or HDFS respectively.
+            Fortunately there's this tool called :py:mod:`luigi` that can
+            help you transfer data from one place to another.
+            """
+        return
+
+    def output(self):
+        """ Output formats supported by MRJob:
+            - directory in local filesystem
+            - directory in HDFS
+            - directory in S3
+
+            Custom implemented output formats:
+
+            - `a single file` in local filesystem (this is a custom feature)
+
+        The last one is implemented using
+        `mrjob.Runner.MRJobRunner.stream_output()`. Because said method
+        returns bytes and not strings and luigi expects strings it is best to
+        just pass format.Nop as the format in this case.
+        """
+
+    def mrjob_opts(self):
+        """ Override this to pass command line options. You probably want to
+        pass a config file (or make one available at the standard locations
+        MRJob reads ) and the runner type.
+        """
+        return {}
+
+    def handle_counters(self, counters):
+        """ Override this if you want to do something with the counters
+        returned by `mrjob.runner.MRJobRunner.counters()`.
+        """
+        return
+
+    def __init__(self, *args, **kwargs):
+        super(MRJobTask, self).__init__(*args, **kwargs)
+        self.opts = self.make_mrjob_opts(*args, **kwargs)
+
+    def run(self):
+        # if the last step fails it's possible for mrjob to
+        # have created the target dir at least on s3
+        # without the actually being successfull. this causes an error and
+        # requires manual intervention. should we just delete the dir on s3 if
+        # it doesn't contain a _SUCCESS ?
```

Comment:
The problem you mention in the inline comments here is the one of atomic movedir. Can you make MRJob always write to a temp-name and in the end rename the file to the original filename if MRJob gives 0 exit code?